velox/functions/sparksql/aggregates/DecimalSumAggregate.h

```diff
@@ -0,0 +1,170 @@
+  // nonNullGroup in writeIntermediateResult to output a null group as sum =
+  // 0, isEmpty = true. nonNullGroup is only available when default-null
+  // behavior is disabled.
+  static constexpr bool default_null_behavior_ = false;
+
+  static bool toIntermediate(
+      exec::out_type<Row<TSumType, bool>>& out,
+      exec::optional_arg_type<TInputType> in) {
+    if (in.has_value()) {
+      out.copy_from(std::make_tuple(static_cast<TSumType>(in.value()), false));
+      return true;
+    }
+    return false;
+  }
+
+  // This struct stores the sum of input values, overflow during accumulation,
+  // and a bool value isEmpty used to indicate whether all inputs are null. The
+  // initial value of sum is 0. We need to keep sum unchanged if the input is
+  // null, as sum function ignores null input. If the isEmpty is true, then it
+  // means there were no values to begin with or all the values were null, so
+  // the result will be null. If the isEmpty is false, then if sum is nullopt
+  // that means an overflow has happened, it returns null.
+  struct AccumulatorType {
+    std::optional<int128_t> sum_{0};
+    int64_t overflow_{0};
+    bool isEmpty_{true};
+
+    AccumulatorType() = delete;
+
+    explicit AccumulatorType(HashStringAllocator* /*allocator*/) {}
+
+    std::optional<int128_t> computeFinalResult() const {
+      if (!sum_.has_value()) {
+        return std::nullopt;
+      }
+      auto adjustedSum =
+          DecimalUtil::adjustSumForOverflow(sum_.value(), overflow_);
+      uint8_t maxPrecision = std::is_same_v<TSumType, int128_t>
+          ? LongDecimalType::kMaxPrecision
+          : ShortDecimalType::kMaxPrecision;
+      if (adjustedSum.has_value() &&
+          DecimalUtil::valueInPrecisionRange(adjustedSum, maxPrecision)) {
+        return adjustedSum;
+      } else {
+        // Found overflow during computing adjusted sum.
+        return std::nullopt;
+      }
+    }
+
+    bool addInput(
+        HashStringAllocator* /*allocator*/,
+        exec::optional_arg_type<TInputType> data) {
+      if (!data.has_value()) {
+        return false;
+      }
+      int128_t result;
+      overflow_ +=
+          DecimalUtil::addWithOverflow(result, data.value(), sum_.value());
+      sum_ = result;
+      isEmpty_ = false;
+      return true;
+    }
+
+    bool combine(
+        HashStringAllocator* /*allocator*/,
+        exec::optional_arg_type<Row<TSumType, bool>> other) {
+      if (!other.has_value()) {
+        return false;
+      }
+      auto otherSum = other.value().template at<0>();
+      auto otherIsEmpty = other.value().template at<1>();
+
+      // IsEmpty should always has value.
+      VELOX_CHECK(otherIsEmpty.has_value());
+
+      bool bufferOverflow = !isEmpty_ && !sum_.has_value();
+      bool inputOverflow = !otherIsEmpty.value() && !otherSum.has_value();
+      if (bufferOverflow || inputOverflow) {
+        sum_ = std::nullopt;
+        return false;
```

Comment:
Returning false means the nullness of this group remains unchanged, e.g., if it was previously a null group, it remains a null group. Since we change the value of sum_ here, shouldn't we return true here?

Reply:
Yes, you are right, I will fix it.